### PR TITLE
[receiver/rabbitmq] Create component

### PR DESCRIPTION
**Pull request by cpheps**
**Description:** Established codebase for a RabbitMQ metric receiver.

- Defined configuration and validation
- Created barebones factory
- Stubbed out `metadata.yaml`
- Updated changelog (will add PR number once it's created)

**Testing:** Units tests for config validation and factory creation.

**Documentation:** Added basic receiver documentation that covers how monitoring is achieved, environment setup, and configuration.

**Use-Case:** Create a metric receiver to monitor a RabbitMQ node, a user can get KPI metrics on their RabbitMQ environment by setting up collectors on each node.

Sponsor: @djaglowski

### CODE REVIEW

**Comment by mx-psi** [comment]
Could you create an issue with the items on #7058?

File: receiver/rabbitmqreceiver/README.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+This receiver supports RabbitMQ versions `3.8` and `3.9`.
+
+The RabbitMQ Management Plugin must be enabled by following the [official instructions](https://www.rabbitmq.com/management.html#getting-started).
+
+Also, a user with at least [monitoring](https://www.rabbitmq.com/management.html#permissions) level permissions must be used for monitoring.
+
+## Configuration
+
+The following settings are required:
+- `endpoint` (default: `http://localhost:15672`): The URL of any node within the cluster to be monitored.
```
[the remark this answers is not in the document]
Comment (cpheps) [reply receiver/rabbitmqreceiver/README.md:19]: Yeah, that's fair. I'll move it to the optional section of the readme.

**Review by mx-psi — APPROVED** [review]
LGTM, needs approval from @djaglowski as a sponsor

**Review by djaglowski — APPROVED** [review]
Looks good to me.

**Comment by djaglowski** [comment]
CI failure was unrelated. Rerunning.

**Pull request merged**